graph/src/components/store.rs

```diff
@@ -660,23 +660,16 @@ where
     /// on the returned stream as a single `StoreEvent`; the events are
     /// combined by using the maximum of all sources and the concatenation
     /// of the changes of the `StoreEvents` received during the interval.
-    pub fn throttle_while_syncing(
+    pub async fn throttle_while_syncing(
         self,
         logger: &Logger,
         store: Arc<dyn QueryStore>,
         interval: Duration,
     ) -> StoreEventStreamBox {
-        // We refresh the synced flag every SYNC_REFRESH_FREQ*interval to
-        // avoid hitting the database too often to see if the subgraph has
-        // been synced in the meantime. The only downside of this approach is
-        // that we might continue throttling subscription updates for a little
-        // bit longer than we really should
-        static SYNC_REFRESH_FREQ: u32 = 4;
-
-        // Check whether a deployment is marked as synced in the store
-        let mut synced = store.is_deployment_synced().unwrap_or(false);
-        let synced_check_interval = interval.checked_mul(SYNC_REFRESH_FREQ).unwrap();
-        let mut synced_last_refreshed = Instant::now();
+        // Check whether a deployment is marked as synced in the store. Note that in the moment a
+        // subgraph becomes synced any existing subscriptions will continue to be throttled since
+        // this is not re-checked.
+        let synced = store.is_deployment_synced().await.unwrap_or(false);
```

Comment:
We close subscriptions after a while, right? So the effect would just be that subscriptions are throttled for a bit longer than they are now, which seems fine.

Reply:
Also, I find querying or subscribing to a subgraph that's still syncing a very questionable thing to do in any sort of prod environment anyways.